File: app/src/main/java/com/example/testhexagongame/piece/PieceI.java
package com.example.testhexagongame.piece;

import com.example.testhexagongame.tiles.tile.Box;
import com.example.testhexagongame.tiles.tile.Shape.Shape;

public interface PieceI<T extends Shape, U> {
    public void create();
    public Boolean putPiece(Box<T, U> current);
    public Boolean isFit(Box<T, U> current);
}

File: app/src/main/java/com/example/testhexagongame/piece/PieceI.kt
package com.example.testhexagongame.piece

import com.example.testhexagongame.tiles.tile.Box
import com.example.testhexagongame.tiles.tile.Shape.Shape

interface PieceI<T: Shape> {
    fun create()
    fun putPiece(current: Box<T>?): Boolean
}

interface PieceFlippable<T: Shape>: PieceI<T> {
    fun flip()
}